templates/util.go

```diff
@@ -106,3 +112,53 @@ func checkNode(node parse.Node, executedTmpls map[string]struct{}) {
 		checkBranchNode(&n.BranchNode, executedTmpls)
 	}
 }
+
+// ParseTestTemplate parses the test template and returns the top-level definitions that should be interpolated as results.
+func ParseTestTemplate(name string, text string) ([]string, error) {
+	var tmpl *tmpltext.Template
+	var capture template.Option = func(text *tmpltext.Template, _ *tmplhtml.Template) {
```

Comment:
NIT: Tempting to get rid of all these awkward `template.Option` hacks used to capture the underlying `tmpltext` at this point by just creating our own `Template` type in `fromContent` that stores a reference to the `tmpltext`. Not necessary though as I imagine the next iteration implementing the factory might change this bit as well.

Reply:
I will see what can be done in the follow-up PRs.